edb/ir/ast.py

```diff
@@ -530,14 +530,28 @@ class TypeRoot(Expr):
     skip_subtypes: bool = False
 
 
-T_co = typing.TypeVar('T_co', covariant=True, bound=typing.Optional[Expr])
+class RefExpr(Expr):
+    '''Different expressions sorts that refer to some kind of binding.'''
+    __abstract_node__ = True
+    typeref: TypeRef
+
+
+class MaterializedExpr(RefExpr):
+    pass
+
+
+class VisibleBindingExpr(RefExpr):
+    pass
```

Comment:
So before we had just `Set.expr = None` and now we have separate subclasses of Expr to be more explicit?

Reply:
Yeah